File: myProblems/R/1.r
# @author R. C. Howell
# @question 1
# @lang R
# @question Find the sum of all the multiples of 3 or 5 below 1000.
# @solution 233168

threes   <- seq(from = 0, to = 999, by = 3)
fives    <- seq(from = 0, to = 999, by = 5)
fifteens <- seq(from = 0, to = 999, by = 15)

sum(sum(threes), sum(fives)) - sum(fifteens)

File: myProblems/C/1.c
// @author R. C. Howell
// @question 1
// @lang C
// @solution 233168

/**
Find the sum of all the multiples of 3 or 5 below 1000.
**/

#include <stdlib.h>
#include <stdio.h>

#define MAX 1000

// naive solution
int main(int argc, char * argv[]) {
    int total = 0;
    int i = 0;
    while ((++i) < MAX) {
        total += (!(i%3) || !(i%5)) ? i : 0;
    }
    printf("%d\n", total);

    return 0;
}